public/scripts/tagsList.js

```diff
@@ -0,0 +1,30 @@
+function getTagObject(content){
+    let tagElement = document.createElement('div')
+    tagElement.classList.add('tagElement')
+
+    let tagElementText = document.createElement('span')
+    let text = document.createTextNode(content)
+    tagElementText.classList.add('tagElementText')
+    tagElementText.appendChild(text)
+
+    let deleteTag = document.createElement('button')
+    deleteTag.value = "X"
+
+    tagElement.appendChild(tagElementText)
+    tagElement.appendChild(deleteTag)
+
+    return tagElement
```

Comment:
Although not incorrect, there are other, less costly [alternatives](https://stackoverflow.com/a/53578013/394589) to generating DOM nodes.

Just an FYI: this way of generating DOM nodes is frowned upon, as it not only couples logic with presentation (JS vs HTML), but it also tends to hinder performance if called often enough. But I understand you're doing this because you only want to do it with vanilla JS. Just remember that JS frameworks were made especially to avoid doing such DOM manipulations with JS (i.e. React, Angular, Vue etc.).

Reply:
I was actually thinking that it's a bit awkward to create things like this and wondering if there is a better way! I will look into it.

Reply:
This was the most fun to fix!

Reply:
Push the fix then, and mark this convo as resolved, using the button below this thread. This way, you mark things as done.

Reply:
Oh, wait, you already did.